mongomock/aggregate.py

```diff
@@ -626,6 +627,61 @@ def _handle_conditional_operator(self, operator, values):
             'aggregation pipeline, it is currently not implemented '
             ' in Mongomock.' % operator)
 
+    def _handle_control_flow_operator(self, operator, values):
+        if operator == '$switch':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$switch requires an object as an argument, '
+                    'found: %s' % type(values)
+                )
+
+            branches = values.get('branches', [])
+            if not isinstance(branches, list):
+                raise OperationFailure(
+                    "$switch expected an array for 'branches', "
+                    'found: %s' % type(branches)
+                )
+            if len(branches) == 0:
+                raise OperationFailure(
+                    '$switch requires at least one branch.'
+                )
+
+            for branch in branches:
+                if not isinstance(branch, dict):
+                    raise OperationFailure(
+                        '$switch expected each branch to be an object, '
+                        'found: %s' % type(branch)
+                    )
+                if 'case' not in branch:
+                    raise OperationFailure(
+                        "$switch requires each branch have a 'case' expression"
```

Comment:
missing a `to` before `have`, same below

Reply:
I understand that omission of `to` in the message is ungrammatical.

I took all error messages from real connection to the MongoDB (version 4.2.6) via pymongo (version 3.10.1, the version 3.11.0 appends `full error:` info) using code similar to:

```py
try:
    collection.aggregate([
        {
            "$project": {
                'new_field': {
                    '$switch': {
                        'branches': [{}],
                    }
                }
            }
        },
    ])
except Exception as exc:
    print(str(exc))
```

This is also why periods on the end of messages are inconsistent:

```
$switch requires each branch have a 'case' expression
$switch requires each branch have a 'then' expression.
```

I am not sure what do do here. Should I make grammatical sentence or keep it just like real MongoDB+pymongo?

Reply:
Both are fine by me (either grammatically correct or like real MongoDB+pymongo). Having the exact same error message is not required because they tend to change it (here we could imagine that they would fix it at one point), so my take on that for now is: make sure to use the same exception type and use an error message that is understandable by a user. For the rest, I'm OK to go with the wind and sometimes be consistant with pymongo, sometimes make a better message.

Leave it as such so that we can get your PR in before the next release (that we should make soon as pymongo v3.11 is incompatible with the latest release of mongomock)